### ISSUE TITLE
Default in-momory cache uses references to store cached responses

### ISSUE THREAD

**Issue by krzysztofslonka** [Schibsted-Tech-Polska/good-guy-http#5]
Default in-momory cache goes against common sense when it comes to multiple request to the same resource. I think that when I use a library that caches the response it is decoupled from anything that I do with it. 

``` javascript

var assert = require('assert');
var gg = require('good-guy-http')();
var got = require('got');
var url = 'http://xkcd.com/info.0.json';

describe('good-guy-http', function() {
  it('should not keep references in in-memory cache', function() {
    var response = gg(url).then(function(response) {
        response.hello = 'no';

        return gg(url)
    }).then(function(response) {
        assert.equal(response.hello, null);
    });

    return response;
  })
})

describe('got', function() {
  it('should not keep references in in-memory cache', function() {
    var response = got(url).then(function(response) {
        response.hello = 'no';

        return got(url)
    }).then(function(response) {
        assert.equal(response.hello, null);
    });

    return response;
  })
})

```

**Comment by krajzeg** [comment]
The in-memory cache stores the actual instance of the object, that's true. It was done this way mostly for performance-related purposes to avoid copying, and I figured it wouldn't hurt much - but I agree that it's counterintuitive.

I guess that's something that could be fixed in the next release, since the principle of least surprise should probably trump saving a few CPU cycles.

**Comment by kjarmicki** [comment]
Released under 1.7.2
